TestSocketProxyd fails on old distributions

The new test TestSocketProxyd was recently implemented by https://github.com/coreos/rkt/pull/2257.

It fails on CentOS 7 and Fedora 22:

```
--- FAIL: TestSocketProxyd (0.25s)
	rkt_socket_proxyd_test.go:192: read tcp [::1]:60169->[::1]:39719: read: connection reset by peer
```

I suspect that those distributions have an old systemd version without systemd-proxyd. The test should be skipped when systemd-proxyd is not installed.

--- a/tests/rkt_socket_proxyd_test.go
+++ b/tests/rkt_socket_proxyd_test.go
@@ -36,6 +36,11 @@ func TestSocketProxyd(t *testing.T) {
 		t.Skip("Systemd is not running on the host.")
 	}
 
+	socketProxydPath := "/lib/systemd/systemd-socket-proxyd"
+	if _, err := os.Stat(socketProxydPath); os.IsNotExist(err) {
+		t.Skip("systemd-socket-proxyd is not installed.")
+	}
+
 	ctx := testutils.NewRktRunCtx()
 	defer ctx.Cleanup()
 
@@ -135,10 +140,11 @@ func TestSocketProxyd(t *testing.T) {
 	After=%s
 
 	[Service]
-	ExecStart=/lib/systemd/systemd-socket-proxyd %s:%d
+	ExecStart=%s %s:%d
 	`
 
-	proxyContent := fmt.Sprintf(proxyToRktTestingEchoService, serviceTargetBase, serviceTargetBase, containerIP, port)
+	proxyContent := fmt.Sprintf(proxyToRktTestingEchoService, serviceTargetBase, serviceTargetBase,
+		socketProxydPath, containerIP, port)
 	proxyContentBase := fmt.Sprintf("proxy-to-rkt-testing-socket-activation-%d.service", rnd)
 	proxyTarget := filepath.Join(unitsDir, proxyContentBase)
 

